packages/react-styled-ui/src/RadioGroup/index.js

```diff
@@ -15,27 +16,40 @@ const RadioGroup = ({
   variantColor,
   onChange,
 }) => {
-  const { current: isControlled } = useRef(valueProp != null);
-  const [value, setValue] = useState(defaultValue || null);
-  const _value = isControlled ? valueProp : value;
-  const _onChange = event => {
-    if (!isControlled) {
-      setValue(event.target.value);
+  const defaultId = useId();
+  name = name ?? `${config.name}:radio-${defaultId}`;
+
+  const [state, setState] = useState({
+    value: valueProp ?? defaultValue,
+  });
+
+  useEffect(() => {
+    if (valueProp !== undefined) {
+      setState({ value: valueProp });
+    }
+  }, [valueProp]);
+
+  const handleChange = event => {
+    const nextValue = event.target.value;
+
+    if (valueProp !== undefined) {
+      setState({ value: valueProp });
+    } else {
+      setState({ value: nextValue });
     }
-    if (onChange) {
-      onChange(event, event.target.value);
+
+    if (typeof onChange === 'function') {
+      onChange(nextValue, event);
```

Comment:
you changed the order of arguments

Reply:
Hi @KKzLEO 

Sorry for causing the trouble.

We will examine what's the best way to have better usage among `CheckboxGroup` and `RadioGroup`. Please stay tuned for further updates.

* `Chakra UI` useCheckboxGroup (https://github.com/chakra-ui/chakra-ui/blob/main/packages/checkbox/src/use-checkbox-group.ts)
    ```js
    /**
     * The callback fired when any children Checkbox is checked or unchecked
     */
    onChange?(value: StringOrNumber[]): void
    ```

* `Chakra UI` useRadioGroup (https://github.com/chakra-ui/chakra-ui/blob/main/packages/radio/src/use-radio-group.ts)
    ```js
    /**
     * Function called once a radio is checked
     * @param nextValue the value of the checked radio
     */
    onChange?(nextValue: string): void
    ```

* `Material UI` RadioGroup (https://github.com/mui-org/material-ui/blob/master/packages/mui-material/src/RadioGroup/RadioGroup.js)
    ```js
    /**
     * Callback fired when a radio button is selected.
     *
     * @param {React.ChangeEvent<HTMLInputElement>} event The event source of the callback.
     * @param {string} value The value of the selected radio button.
     * You can pull out the new value by accessing `event.target.value` (string).
     */
    onChange: PropTypes.func,
    ```

cc @tinaClin

Reply:
After internal discussion, we will keep the new breaking change introduced in v0.19.0 and remove the second `event` argument since the v1 release.

I have highlighted the breaking change in the release notes: https://github.com/trendmicro-frontend/tonic-ui/releases/tag/%40trendmicro%2Freact-styled-ui%400.19.0

#### Radio Example

```jsx
function RadioExample() {
  const [value, setValue] = React.useState('1')

  return (
    <RadioGroup onChange={setValue} value={value}>
      <Stack direction='row'>
        <Radio value='1'>First</Radio>
        <Radio value='2'>Second</Radio>
        <Radio value='3'>Third</Radio>
      </Stack>
    </RadioGroup>
  )
}
```

Reply:
thanks for your reply and contributions, we've fixed the code